ddl/column_type_change_test.go

```diff
@@ -2244,53 +2244,21 @@ func TestChangeFromTimeToYear(t *testing.T) {
 
 // Fix issue: https://github.com/pingcap/tidb/issues/26292
 // Cast date to timestamp has two kind behavior: cast("3977-02-22" as date)
-// For select statement, it truncate the string and return no errors. (which is 3977-02-22 00:00:00 here)
-// For ddl reorging or changing column in ctc, it need report some errors.
+// For select statement, it truncates the string and return no errors. (which is 3977-02-22 00:00:00 here)
+// For ddl reorging or changing column in ctc, it needs report some errors.
 func TestCastDateToTimestampInReorgAttribute(t *testing.T) {
-	store, dom, clean := testkit.CreateMockStoreAndDomain(t)
+	store, clean := testkit.CreateMockStore(t)
 	defer clean()
 	tk := testkit.NewTestKit(t, store)
-	tk.MustExec("use test;")
+	tk.MustExec("use test")
 
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("CREATE TABLE `t` (`a` DATE NULL DEFAULT '8497-01-06')")
 	tk.MustExec("insert into t values(now())")
 
-	originalHook := dom.DDL().GetHook()
-	defer dom.DDL().(ddl.DDLForTest).SetHook(originalHook)
-
-	// use new session to check meta in callback function.
-	internalTK := testkit.NewTestKit(t, store)
-	internalTK.MustExec("use test")
-
-	tbl := tk.GetTableByName("test", "t")
-	require.NotNil(t, tbl)
-	require.Equal(t, 1, len(tbl.Cols()))
-
-	hook := &ddl.TestDDLCallback{Do: dom}
-	var (
-		checkErr1 error
-		checkErr2 error
-	)
-	hook.OnJobRunBeforeExported = func(job *model.Job) {
```

Comment:
Why do you remove the hook?

Reply:
Let me try to revert this change. According to the comment of the test, either test with ddl hook or serially execute sql statement should be the same.

Reply:
Done by 893d865a441ae4708b8ac28ffd62db04faaa643a